### PR TITLE
Edit and expand SOAP template comments

### PR DESCRIPTION
**Pull request by jonathanalgar**
See https://outsystemsrd.atlassian.net/browse/TK-10028 for more information.

### CODE REVIEW

File: templates/soap/ConsumeSOAPExample/ConsumeSOAPExample.cs [hunk 1]
```diff
@@ -5,32 +5,49 @@
...
+        /// and returns the result in the SOAP response.
+        /// This method is exposed as a server action to your ODC apps and libraries.
+        /// </summary>
         [OSAction(Description = "The sum method takes two numbers as input parameters and returns their sum.", IconResourceName = "ConsumeSOAPExample.Resources.SOAP.jpg", ReturnName = "sum")]
         int Sum(
             [OSParameter(Description = "Number A & B as an integer")] Structures.Numbers numbers
         );
-        // This method is exposed as a server action to your ODC apps and libraries.
+        /// <summary>
+        /// Takes two numbers as input parameters and calculates the difference between
```
Comment (kakabisht) [inline templates/soap/ConsumeSOAPExample/ConsumeSOAPExample.cs:27]: Suggested change: and returns the difference between

File: templates/soap/ConsumeSOAPExample/ConsumeSOAPExample.cs [hunk 2]
```diff
@@ -47,34 +64,45 @@ public Calculator()
...
+                // automatically transforms the method call into a SOAP request, sends the 
+                // request to the web service, and converts the SOAP response back into a .NET object.
                 var result = await proxy.AddAsync(numbers.NumberA, numbers.NumberB);
                 return result;
             }
         }
         public int Subtract(Numbers numbers)
         {
-            /* This line calls the AddAsync method (defined later in the code), which asynchronously calls a SOAP service to add NumberA and NumberB.Once the result is returned, it is stored in the sum variable. */
+            // Creates and uses a SOAP service client (proxy) for the Subtract operation.
```
Comment (kakabisht) [inline templates/soap/ConsumeSOAPExample/ConsumeSOAPExample.cs:88]: Suggested change: public int Sum and  Subtract should have similar comments.

File: templates/soap/ConsumeSOAPExample/ConsumeSOAPExample.cs (hunk 1 above)
Comment (jonathanalgar) [reply templates/soap/ConsumeSOAPExample/ConsumeSOAPExample.cs:27]: Agred, changed.

**Comment by kakabisht** [comment]
@jonathanalgar , I checked the zip file, but I don't know about a couple of lines in the Reference. cs under CalculatorWebServiceSoapClient, Are they being interpreted as comments or code?

- /// <param name="serviceEndpoint">The endpoint to configure</param>
-  /// <param name="clientCredentials">The client credentials</param>

**Comment by jonathanalgar** [comment]
> @jonathanalgar , I checked the zip file, but I don't know about a couple of lines in the Reference. cs under CalculatorWebServiceSoapClient, Are they being interpreted as comments or code?
> 
> * /// The endpoint to configure
> * /// The client credentials

Comments. You can learn about how `param` tags can be used in XML C# comments [here](https://www.pluralsight.com/guides/using-param-paramref-tag-in-csharp-xml-documentation-comments), for example. But I didn't edit the `reference.cs` file in any case—it is automatically generated.

From the perspective of zip the important question is if the `reference.cs` is the same as the original: I believe it is, but please double-check.

**Comment by kakabisht** [comment]
> > @jonathanalgar , I checked the zip file, but I don't know about a couple of lines in the Reference. cs under CalculatorWebServiceSoapClient, Are they being interpreted as comments or code?
> > 
> > * /// The endpoint to configure
> > * /// The client credentials
> 
> Comments. You can learn about how `param` tags can be used in XML C# comments [here](https://www.pluralsight.com/guides/using-param-paramref-tag-in-csharp-xml-documentation-comments), for example. But I didn't edit the `reference.cs` file in any case—it is automatically generated.
> 
> From the perspective of zip the important question is if the `reference.cs` is the same as the original: I believe it is, but please double-check.

I did check.  They are comments.

**Comment by jonathanalgar** [comment]
Approved by @OSmvicente on Slack 🚀

**Pull request merged**